snow/engine/snowman/transitive_test.go

```diff
@@ -2762,54 +2767,39 @@ func TestEngineBubbleVotesThroughInvalidBlock(t *testing.T) {
 	reqID := new(uint32)
 	sender.SendGetF = func(_ context.Context, inVdr ids.NodeID, requestID uint32, blkID ids.ID) {
 		*reqID = requestID
-		if *asked {
-			t.Fatalf("Asked multiple times")
-		}
-		if blkID != blk1.ID() {
-			t.Fatalf("Expected engine to request blk1")
-		}
-		if inVdr != vdr {
-			t.Fatalf("Expected engine to request blk2 from vdr")
-		}
+		require.False(*asked)
+		require.Equal(blk1.ID(), blkID)
+		require.Equal(vdr, inVdr)
 		*asked = true
 	}
-	// Receive Gossip message for [blk2] first and expect the sender to issue a Get request for
-	// its ancestor: [blk1].
-	if err := te.Put(context.Background(), vdr, constants.GossipMsgRequestID, blk2.Bytes()); err != nil {
-		t.Fatal(err)
-	}
-
-	if !*asked {
-		t.Fatalf("Didn't ask for missing blk1")
-	}
+	// this engine receives Gossip message for [blk2] which is "unknown" in this engine
+	// thus sending a Get request for its ancestor [blk1]
+	// (see above for expected "Get" request)
+	require.NoError(te.Put(context.Background(), vdr, constants.GossipMsgRequestID, blk2.Bytes()))
+	require.True(*asked)
 
 	// Prepare to PushQuery [blk1] after our Get request is fulfilled. We should not PushQuery
 	// [blk2] since it currently fails verification.
 	queried := new(bool)
 	queryRequestID := new(uint32)
 	sender.SendPushQueryF = func(_ context.Context, inVdrs set.Set[ids.NodeID], requestID uint32, blkBytes []byte) {
-		if *queried {
-			t.Fatalf("Asked multiple times")
-		}
+		require.False(*queried)
 		*queried = true
+
 		*queryRequestID = requestID
 		vdrSet := set.Set[ids.NodeID]{}
 		vdrSet.Add(vdr)
-		if !inVdrs.Equals(vdrSet) {
-			t.Fatalf("Asking wrong validator for preference")
-		}
-		if !bytes.Equal(blk1.Bytes(), blkBytes) {
-			t.Fatalf("Asking for wrong block")
-		}
+		require.Equal(vdrSet, inVdrs)
+		require.Equal(blk1.Bytes(), blkBytes)
 	}
+	// this engine now answers the "Get" request, and this should allow [blk1] to be issued
+	// and cause [blk2] to fail verification.
+	// (see above for expected "PushQuery" request)
```

Comment:
The engine is actually processing the _response_ to the "Get" request (not the "Get" request itself).
```suggestion
	// This engine now handles the response to the "Get" request. This should cause [blk1] to be issued
	// which will result in attempting to issue [blk2]. However, [blk2] should fail verification and be dropped.
	// By issuing [blk1], this node should fire a "PushQuery" request for [blk1].
	// (see above for expected "PushQuery" request)
```